### ISSUE TITLE
Update query depth limiter ?

### ISSUE THREAD

**Issue by patrick91** [strawberry-graphql/strawberry#2338]
> In the function which checks which field should be ignored by the query depth limiter, you should pass the field object instead of field name. This way we can do more checking with field attributes like its arguments if needed

we received this on discord

**Comment by tsmith023** [comment]
I can take a stab at this one @patrick91, unless someone else has claimed it on discord already!

**Comment by patrick91** [comment]
@tsmith023 go for it 😊

**Comment by tsmith023** [comment]
@XChikuX, yep updates to the docs is part of the PR. It details how to use the new dataclass object!

**Comment by patrick91** [comment]
Done in https://github.com/strawberry-graphql/strawberry/pull/2505
